File: C++/Mathematics/palindrome-check.cpp
#include <iostream>
#include <limits.h>

using namespace std;

string isPalindrome(int number) {
    int rev = 0;
    int temp = number;

    while(temp != 0) {
        int ld = temp % 10;
        rev = rev * 10 + ld;
        temp = temp / 10;
    }

    if (rev == number) {
        return std::to_string(number) + " is Palindrome!";
    } else {
        return std::to_string(number) + " is not Palindrome!";
    }
}

int main() {
    cout << isPalindrome(123454321) << "\n";
    cout << isPalindrome(1231) << "\n";
}

File: JavaScript/Mathematics/palindrome-check.js
function isPalindromeJS(number) {
    let rev = 0;
    let temp = number;

    while(temp != 0) {
        let ld = temp % 10;
        rev = rev * 10 + ld;
        temp = (temp - temp % 10) / 10;
    }

    if (rev == number) {
        return number + " is Palindrome!";
    } else {
        return number + " is not Palindrome!";
    }
}

console.log(isPalindromeJS(123454321) + "\n");
console.log(isPalindromeJS(1231) + "\n");